src/main/java/net/snowflake/client/jdbc/SnowflakeResultSetMetaDataV1.java

```diff
@@ -96,7 +97,23 @@ public boolean isAutoIncrement(int column) throws SQLException {
 
   @Override
   public boolean isCaseSensitive(int column) throws SQLException {
-    return false;
+    int colType = getColumnType(column);
+
+    switch (colType) {
+      case Types.INTEGER:
+      case Types.BIGINT:
+      case Types.DECIMAL:
+      case Types.DOUBLE:
+      case Types.BOOLEAN:
+      case Types.TIMESTAMP:
+      case Types.TIMESTAMP_WITH_TIMEZONE:
+      case Types.DATE:
+      case Types.TIME:
+        return false;
+
+      default:
+        return true;
```

Comment:
Can we explicit mention other types instead of "default" for true? Then default becomes false. So two lists, known data types for false, known data types for true and default as false.

Reply:
done, thanks!